### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by Uzair-Manzoor**
## This pull request has been created against project accomplishment as per given instructions, below mentioned implementations have been made step-by-step:

- [x] Developed a React application
- [x] Installed project dependencies
- [x] Implemented JavaScript and styling linters for code quality
- [x] Created the main application component
- [x] Configured and exported the Redux store
- [x] Incorporated application routes within the App component
- [x] Integrated APIs for weather and air quality metrics
- [x] Handled error messages within the application
- [x] Added real-time filtering via a search bar on the States and Cities pages
- [x] Implemented unit tests for pure functions within Redux code
- [x] Conducted integration tests for the application using React Testing Library
- [x] Ensured API access is mocked to prevent actual requests during testing

### CODE REVIEW

File: LICENSE
Comment (AbelG101) [inline LICENSE:1]: - I'd like to start off by saying well done in building your project here it's fully functional and works like a charm but there are some minor issues I discovered so please take carefull note of them.
- As part of the project requirements please make sure to record a video that contains:
  - [x] A maximum length of 5 minutes.
  - [x] A clear and objective description of the project in professional English. The topic and purpose of the project should be made clear.
  - [x] A demo of the project, highlighting all the project features.
  - [x] A highlight about a piece of the code that you find interesting or proud of making.
- [x] Also make sure to include a link to a live demo of your web app in the readme.

File: src/App.css
Comment (AbelG101) [inline src/App.css:1]: - [x] Good job using the "Lato" font for the elements in your design but also make sure to use the "Gill sans font" as specified in the Behance design guideline for the body and the title of your UI.
- [x] I've also noticed uniform or similar font-weights and font-size in your UI which is creating a discrepancy when compared with the original Behance design so I'd suggest you revise your design and make the necessary adjustments

Comment (AbelG101) [inline src/App.css:1]: - [x] As part of the project requirements I'd suggest you improve the layout: composition and space between elements of your design to ensure compliance with the original Behance design.
- Your design:
  image
  image
- Original design:
    image

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @Uzair-Manzoor 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- App functions properly. ✔
- The home page and detail page work as expected. ✔
- Correct use of React router ✔
- Properly written Unit & Integration tests ✔
- Correct use of Git Flow ✔
- Well-structured files ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi @Uzair-Manzoor, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


### Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**